extension/httpforwarderauthextension/extension.go

```diff
@@ -77,34 +79,70 @@ func (e *authExtension) authenticate(ctx context.Context, headers map[string][]s
 	if len(authHeaders) == 0 || authHeaders[0] == "" {
 		return ctx, errNotAuthenticated
 	}
+	apikey := authHeaders[0]
+	var err error
+	if e.cfg.Enable && e.cfg.SecretKey != "" {
+		apikey, err = AesDecrypt(authHeaders[0], e.cfg.SecretKey, e.cfg.IV)
+		if err != nil {
+			e.logger.Error("[httpforwarderauthextension] aes decrypt error: ", zap.Error(err))
+			return ctx, errCheckErrAuthentication
+		}
+	}
+
+	// extend{"authentication":"xx", "custom_tag1":"xx", "custom_tag2":"xx"}
+	// authentication is required, custom tags will be added to span tags
+	if strings.HasPrefix(apikey, ExtendAuthenticationPrefix) {
+		split := strings.Split(apikey, ExtendAuthenticationPrefix)
+		m := make(map[string]string)
+		err = json.Unmarshal([]byte(split[1]), &m)
+		if err != nil {
+			e.logger.Error("[httpforwarderauthextension] extend authentication unmarshal error: ", zap.Error(err))
+			return nil, err
+		}
+		apikey = m[Authentication]
+		delete(m, Authentication)
+		extendTags, err := json.Marshal(m)
+		if err != nil {
+			e.logger.Error("[httpforwarderauthextension] extend tags marshal error: ", zap.Error(err))
+			return nil, err
+		}
+		headers[ExtendTags] = []string{string(extendTags)}
```

Comment:
Put extendTags at grpc context to avoid serialization and deserialization.

Reply:
i have fixed it